rllib/offline/dataset_reader.py

```diff
@@ -114,20 +147,36 @@ def __init__(self, ioctx: IOContext, ds: ray.data.Dataset):
             ds: Ray dataset to sample from.
         """
         self._ioctx = ioctx
+        self._default_policy = self.policy_map = None
         self._dataset = ds
+        self.count = None if not self._dataset else self._dataset.count()
+        # do this to disable the ray data stdout logging
+        ray.data.set_progress_bars(enabled=False)
+
         # the number of rows to return per call to next()
         if self._ioctx:
-            self.batch_size = ioctx.config.get("train_batch_size", 1)
-            num_workers = ioctx.config.get("num_workers", 0)
+            self.batch_size = self._ioctx.config.get("train_batch_size", 1)
+            num_workers = self._ioctx.config.get("num_workers", 0)
+            seed = self._ioctx.config.get("seed", None)
+            if seed and not isinstance(seed, int):
+                raise ValueError(
+                    "If a random seed is specified, seed can only be an "
+                    "integer type."
+                )
         if num_workers:
             self.batch_size = max(math.ceil(self.batch_size / num_workers), 1)
         # We allow the creation of a non-functioning None DatasetReader.
         # It's useful for example for a non-rollout local worker.
         if ds:
+            if self._ioctx.worker is not None:
+                self._policy_map = self._ioctx.worker.policy_map
+                self._default_policy = self._policy_map.get(DEFAULT_POLICY_ID)
```

Comment:
man, input_reader shouldn't have access to policy. this seems bad.
let's either find a different place for it, or make sure the data file has all the post-processed data bits in the first place, so we don't have to run it?

Reply:
the issue is that we need a way to support doing do trajectory post processing if needed and the only way to do that is if I have access to the policy. 

This is how its done in the JSON reader I really just copied the logic. I would remove this, but needed to do this to pass tests. @gjoliver

Reply:
yeah I see why you are doing this. During sampling env_runner calls this method and since now we are replacing env_runner basically with this input_reader we have to do something equivalent. It's a bad design for both actually. The post_processing trajectory should get called by default where the post_fn function is getting called right now. All these can be addressed with a proper callback design but we are far from that right now.